### PR TITLE
Remove _mean when calculating sampler acceptance

### LINKED ISSUE #468: Getting acceptance of numpy metropolis_hastings
The method defining the acceptance for `numpy/metropolis_hastings` in v3.0 is

```python
@property
def acceptance(self):
    """The measured acceptance probability."""
    return _mean(self._accepted_samples) / _mean(self._total_samples)
```

But `self._accepted_samples` and `self._total_samples` are ints. Don't know if this has been fixed in a PR, but if it's not, maybe it should be included.

### PR DESCRIPTION
**Pull request by VolodyaCO**
Closes #468

### CODE REVIEW

**Comment by gcarleo** [comment]
But removing _mean will cause issues when using MPI, since each processor will have its own acceptance. Can we use _sum instead of mean and cast to float if that was the original issue?

**Comment by PhilipVinc** [comment]
@VolodyaCO what exactly is the error when `_mean` is included?

**Comment by PhilipVinc** [comment]
If it's erroring within `_sum_inplace` by saying `Unknown type to perform dispatch upon` it's a very easy fix:
In `netket/netket/stats/_sum_inplace.py:28`

```python
#######
# Scalar
@sum_inplace.register(complex)
@sum_inplace.register(_np.float64)
@sum_inplace.register(_np.float32)
@sum_inplace.register(_np.complex64)
@sum_inplace.register(_np.complex128)
@sum_inplace.register(float)
@sum_inplace.register(int)                              ### <---- Add this line
def sum_inplace_scalar(a):
    ar = _np.asarray(a)

```

**Comment by VolodyaCO** [comment]
> @VolodyaCO what exactly is the error when `_mean` is included?

```bash
return _mean(self._accepted_samples) / _mean(self._total_samples)
  File "/home/vladimir/.local/share/virtualenvs/salesman_problem_netket-UkBmDymc/lib/python3.7/site-packages/netket/stats/mpi_stats.py", line 37, in mean
    out = a.mean(axis=axis)
AttributeError: 'int' object has no attribute 'mean'
```

**Comment by VolodyaCO** [comment]
I just deleted the mean in `numpy/metropolis_hastings` because for those samplers the `self._accepted_sample`s and `self._total_samples` are ints. And `_mean ` is not used elsewhere

File: netket/sampler/numpy/metropolis_hastings.py [hunk 1]
```diff
@@ -127,4 +126,4 @@ def __next__(self):
     @property
     def acceptance(self):
         """The measured acceptance probability."""
-        return _mean(self._accepted_samples) / _mean(self._total_samples)
```
Comment (PhilipVinc) [inline netket/sampler/numpy/metropolis_hastings.py:130]: I think the correct fix would be

```
    def acceptance(self):		     def acceptance(self):
         """The measured acceptance probability."""		         """The measured acceptance probability."""
         return _sum_inplace(self._accepted_samples) /self._total_samples		
```

Comment (gcarleo) [reply netket/sampler/numpy/metropolis_hastings.py:130]: total_samples is already summed over MPI nodes?

Comment (PhilipVinc) [reply netket/sampler/numpy/metropolis_hastings.py:130]: Together with register the dispatch rule that I posted above

Comment (PhilipVinc) [reply netket/sampler/numpy/metropolis_hastings.py:130]: @gcarleo nope, you are right. I fixed it

Comment (VolodyaCO) [reply netket/sampler/numpy/metropolis_hastings.py:130]: Ok, I cannot modify or approve this suggestion. Also, _sum_inplace has to be imported from stats, right?

**Review by PhilipVinc — APPROVED** [review]

**Comment by PhilipVinc** [comment]
@gcarleo can you merge this (only macOS black is failing)

**Pull request merged**